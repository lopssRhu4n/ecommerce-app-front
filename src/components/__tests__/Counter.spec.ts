import { describe, it, expect } from 'vitest';
import { mount } from '@vue/test-utils';
import Counter from '@/components/CounterComponent.vue';

describe('Counter', () => {
  const wrapper = mount(Counter);
  it('should render correctly', () => {
    expect(wrapper.html()).toContain('Current value: 0');
  });

  it('should increment correctly', async () => {
    await wrapper.find('[data-type="increment"]').trigger('click');
    expect(wrapper.html()).toContain('1');
  });

  it('should decrement correctly', async () => {
    await wrapper.find('[data-type="decrement"]').trigger('click');
    expect(wrapper.html()).toContain('-1');
  });
});
