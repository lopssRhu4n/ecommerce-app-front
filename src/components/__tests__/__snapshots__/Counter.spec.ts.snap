// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Counter > should render correctly 1`] = `
"<p>0</p>
<button>Increment</button>
<button>Decrement</button>"
`;
